### PR TITLE
VIDEO-8868 Connect to a RSP v3 Room (Large Room)

### PR DESCRIPTION
**Pull request by manjeshbhargav**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### Description

Changes:

1. Set RSP version to 3.
2. Always request track_subscriptions MSP.

### Before review
* [ ] Updated CHANGELOG.md if necessary
* [ ] Added unit tests if necessary
* [ ] Updated affected documentation
* [ ] Verified locally with `npm test`
* [ ] Manually sanity tested running locally
* [ ] Included screenshot as PR comment (if needed)
* [ ] Ready for review

### CODE REVIEW

**Review by cmebarrow — APPROVED** [review]
LGTM. Just one point: will the code stay under lib/signaling/v2 or move under v3? I guess that'll be sorted out later.

**Comment by manjeshbhargav** [comment]
@cmebarrow 
> LGTM. Just one point: will the code stay under lib/signaling/v2 or move under v3? I guess that'll be sorted out later.
Yes. This will happen in future PRs.

File: lib/util/index.js [hunk 1]
```diff
@@ -718,7 +718,9 @@ function createMediaSignalingPayload(dominantSpeaker, networkQuality, trackPrior
     trackSwitchOff
       // eslint-disable-next-line
       ? { track_switch_off: transports }
-      : {}
+      : {},
+    // eslint-disable-next-line
+    { track_subscriptions: transports }
```
Comment (ceaglest) [inline lib/util/index.js:723]: Does the v3 to v2 downgrade imply that trackSwitchOff is no longer optional? The seamless downgrade is one of the reasons I proposed to make bandwidth profiles default / mandatory in the next major version.

In C++ SDK the track_switch_off is only negotiated only when a bandwidth profile is set today. If we make the API change the implementation would take care of itself I guess.

Comment (manjeshbhargav) [reply lib/util/index.js:723]: @ceaglest Sure, I agree with making bandwidth profile mandatory (plus removing `disabled` for `trackSwitchOffMode`). However, I want to keep the bandwidth profile related changes for large rooms out of scope for March 1. I'm sure the react app always enables bandwidth profile right now (which will be used for the demo).

Comment (ceaglest) [reply lib/util/index.js:723]: Fair enough, as long as we don't expect the final behavior downgrade behavior in the pilot. 👍

File: test/unit/spec/signaling/v2/twilioconnectiontransport.js [hunk 2]
```diff
@@ -182,7 +183,7 @@ describe('TwilioConnectionTransport', () => {
           assert.deepEqual(message.peer_connections, test.peerConnectionManager.getStates());
           assert.equal(message.token, test.accessToken);
           assert.equal(message.type, 'connect');
-          assert.equal(message.version, 2);
+          assert.equal(message.version, 3);
```
Comment (ceaglest) [inline test/unit/spec/signaling/v2/twilioconnectiontransport.js:186]: It looks like the Ice request below is for version 1 (line 200). If possible all the messages should reflect version 3. I don't think it's mandatory for this PR though.

**Review by ceaglest — APPROVED** [review]
Looks good based upon the requirements of the ticket. My comments can definitely be addressed later.

**Pull request merged**